### PR TITLE
WID-40 Add TimeSeriesWidget based on MNE library

### PR DESCRIPTION
**Pull request by liadomide**
Add TimeSeriesWidget based on MNE library
Add Notebooks to test this new widget

### CODE REVIEW

File: tvbwidgets/ui/ts_widget.py [hunk 1]
```diff
@@ -0,0 +1,339 @@
...
+            ('p', 'Toggle draggable annotations'),
+            ('s', 'Toggle scalebars'),
+            ('z', 'Toggle scrollbars'),
+            ('esc', 'Close focused figure or dialog window'),
+            ('MOUSE INTERACTION', ' '),
+            (f'Left-click channel name', 'Mark/unmark bad channel'),
+            (f'Left-click channels data', 'Mark/unmark bad channel'),
+            ('Left-click-and-drag on plot', 'Add annotation (in annotation mode)'),
+            ('Left-click on plot background', 'Place vertical guide'),
+            ('Right-click on plot background', 'Clear vertical guide'),
```
Comment (liadomide) [inline tvbwidgets/ui/ts_widget.py:167]: This relatively long text-based content would rather stay in a separate file. Maybe .ini or .properties ?

File: tvbwidgets/ui/ts_widget.py [hunk 2]
```diff
@@ -0,0 +1,310 @@
...
+
+    # ====================================== CONFIGURATION =============================================================
+    def configure_ts_widget(self):
+        if isinstance(self.data, TimeSeries):
+            self.data_type = TimeSeries
+            self.configure_ch_names_ts()
+            self.configure_displayed_period_ts()
+            self.configure_ch_order_ts()
+            self.configure_ch_types_ts()
+            self.create_raw_from_ts()
```
[the remark this answers is not in the document]
Comment (liadomide) [reply tvbwidgets/ui/ts_widget.py:85]: These `*BuilderFrom*` classes could stay under the same python module `ui/ts_widget.py`, I think...

File: tvbwidgets/api.py [hunk 3]
```diff
@@ -8,4 +8,4 @@
 from .ui.phase_plane_widget import PhasePlaneWidget
 from .ui.storage_widget import StorageWidget
 from .ui.surface_widget import SurfaceWidget
-from .ui.ts_widget import TimeSeriesWidget
+from .ui.ts_widget import TimeSeriesWidget, TSWidgetBuilderFromTVB, TSWidgetBuilderFromNumpy, TSWidgetFactory
```
Comment (liadomide) [inline tvbwidgets/api.py:11]: I think here only `TSWidgetFactory` and `TimeSeriesWidget` are necessary. or ?

**Review by popaula937 — COMMENTED** [review]
Code looks good to me, I see a bit of a difference in the API in contrast with SurfaceWidget in the usage of _get_widget_ which is not necessary for SurfaceWidget (but we could use it for the sake of consistency).

**Comment by liadomide** [comment]
> Code looks good to me, I see a bit of a difference in the API in contrast with SurfaceWidget in the usage of _get_widget_ which is not necessary for SurfaceWidget (but we could use it for the sake of consistency).

Good observation.
The API on `SurfaceWidget` is the desired form. What we have here in `TSWidget` (also in `PhasePlaneWidget`) I would name: a tolerated api for now, but not desired.
I admit not having looked too deeply in this part, I merely have few ideas how the mne figure could be used differently, but didn't yet tested in the code.... I will add at least a TODO on this

File: tvbwidgets/ui/ts_widget.py [hunk 4]
```diff
@@ -289,7 +313,6 @@ def _create_checkboxes(self, array_wrapper):
         unselect_all_btn.on_click(self._unselect_all)
 
         # select dimensions region
-        # TODO position for Sv and mode should come from Wrapper
```
Comment (romina1601) [inline tvbwidgets/ui/ts_widget.py]: As discussed, the user will be instructed that the input should always have the State variable on position 1 and the Mode on position 3, so there is no need to compute their positions in the Wrapper.

**Pull request merged**